solidity/test/VendingMachineTest.js

```diff
@@ -626,4 +667,83 @@ describe("VendingMachine", async function() {
       )
     })
   })
+
+  describe("getMaxSupply", async () => {
+    beforeEach(async () => {
+      await createSnapshot()
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    it("has a max supply of 2 on the first day", async () => {
+      let maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(2))
+
+      await increaseTime(23.5 * 60 * 60) // 23.5 hours
+
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(2))
+
+      await increaseTime(60 * 60) // 1 hour
+
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.not.eq.BN(btcToTbtc(2))
+    })
+
+    it("has a max supply of 100 BTC between the first day and 30th day", async () => {
+      await increaseTime(24 * 60 * 60 + 1) // 1 day and 1 second
+      let maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(100))
+
+      await increaseTime(15 * 24 * 60 * 60) // 15 days
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(100))
+
+      await increaseTime(14 * 24 * 60 * 60 - 10 * 600) // 14 days minus 10 minutes
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(100))
```

Comment:
You test that past the end time the value is not the expected one in the previous test. Why not do that here as well?